sale_report_wizard/wizard/sale_report_wizard.py

```diff
@@ -0,0 +1,61 @@
+import ast
+
+from odoo import api, fields, models
+
+
+class SaleReportWizard(models.TransientModel):
+    _name = "sale.report.wizard"
+    _description = "Sales Report Wizard"
+
+    start_date = fields.Date(string="Start Date")
+    end_date = fields.Date(string="End Date")
+
+    filter_date_type = fields.Selection(
+        [("confirmation_date", "Confirmation_date (keep only confirmed orders)"), 
+        ("date_order", "Order Date")],
+        default="confirmation_date",
+        string="Filter by",
+    )
+
+    report = fields.Many2one("sale.reports.config", string="Preset Report")
+
+    @api.multi
+    def button_show(self):
+        self.ensure_one()
+
+        view_id = self.env.ref(self.report.view).id
+        search_view_id = ""
+        if self.report.search_view:
+            search_view_id = self.env.ref(self.report.search_view).id
+
+        domain = []
+
+        # sale.report name date_order as date,
+        # while report.all.channel.sale and sale.order 
+        # name it as date_order
```

Comment:
Is there other things that make report.all.channel.sale incompatible? Otherwise, your comment in DESCRIPTION.rst is outdated.

Reply:
Ah, yes. It lacked an overwrite of the report.all.channel.sale query to work. It was in the custom module rotordc_sale_reports_top_category. 
So:
- I also added it here.
- Removed the comment in Description.